### DISCUSSION TITLE
Consuming foreign messages

### DISCUSSION THREAD

**Discussion by Bilge** [symfony/symfony#53198] [Q&A]
Is it possible to use a Symfony's Messenger component worker (`messenger:consume`) to consume messages not created by Symfony? It seems to me Messenger, and in particular its `Worker` class, is hard-coded to consume `Envelope`s, which of course would not be created by the foreign system.

**Comment by stefangr** [comment]
You have to create a [message serializer](https://symfony.com/doc/current/messenger.html#message-serializer-for-custom-data-formats) for that.
The serializer will convert the foreign message to an Envelope.

**Comment by Bilge** [comment]
I really don't want to convert anything to an `Envelope`, though, since it accomplishes nothing: it's purely wasted CPU cycles. I just wanted to use Messenger so I can take advantage of the proper signal handling for graceful termination.
